<launch>
	<!-- rviz -->
	<node pkg="rviz" type="rviz" name="rviz" args="-d $(find t_intern_assignment)/rviz_config/road_extraction.rviz"/>

	<!-- data reconstruction -->
	<node pkg="t_intern_assignment" name="data_reconstruction" type="data_reconstruction" output="screen">
		<param name="publish_rate" type="double" value="1"/>
		<param name="child_frame_name" type="string" value="/velodyne"/>
		<param name="parent_frame_name" type="string" value="/odom"/>
		<param name="file_path" type="string" value="/home/amsl/ozaki/job_hunting/toyota/assignment/datasets/bike_bicycle[5]/1"/>
	</node>

	<!-- main -->
	<include file="$(find velodyne_pointcloud)/launch/32e_points.launch"/>
    <node pkg="nodelet" type="nodelet" name="heightmap_nodelet"
        args="load velodyne_height_map/HeightMapNodelet velodyne_nodelet_manager">
        <param name="cell_size" value="0.25"/>
        <param name="full_clouds" value="true"/>
        <param name="grid_dimensions" value="400"/>
        <param name="height_threshold" value="0.1"/>
		<remap from="/velodyne_points" to="/cloud"/>
    </node>
	<node pkg="t_intern_assignment" name="dynamic_pc_extraction_octree_buffer" type="dynamic_pc_extraction_octree_buffer" output="screen">
		<remap from="/cloud" to="/velodyne_obstacles"/>
		<remap from="/cloud/dynamic" to="/velodyne_obstacles/dynamic"/>
		<param name="child_frame_name" type="string" value="/velodyne"/>
		<param name="parent_frame_name" type="string" value="/odom"/>
		<param name="voxel_size" type="double" value="2.0"/>
		<param name="min_points_per_leaf" type="int" value="10"/>
		<param name="buffer_size" type="int" value="10"/>
		<param name="threshhold_dynamic_likelihood" type="double" value="0.6"/>
	</node>
	<node pkg="clustering_pcl" name="euclidean_clustering" type="euclidean_clustering" output="screen">
		<remap from="/velodyne_points" to="/velodyne_obstacles/dynamic"/>
		<param name="cluster_tolerance" type="double" value="0.2"/>
		<param name="min_cluster_size" type="int" value="30"/>
	</node>
</launch>
